mne/viz/evoked.py

```diff
@@ -1705,13 +1741,68 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
                                  "conditions. Condition " + style_ +
                                  " was not found in the supplied data.")
 
-    # second, color
+    # third, color
     # check: is color a list?
     if (colors is not None and not isinstance(colors, string_types) and
             not isinstance(colors, dict) and len(colors) > 1):
         colors = dict((condition, color) for condition, color
                       in zip(conditions, colors))
 
+    if cmap is not None:
+        if not isinstance(cmap, string_types) and len(cmap) == 2:
+            cmap_label, cmap = cmap
+        else:
+            cmap_label = ""
+
+    # dealing with a split legend
+    if split_legend is None:
+        split_legend = cmap is not None  # default to True iff cmap is given
+    if split_legend is True:
+        if colors is None:
+            raise ValueError(
+                "If `split_legend` is True, `colors` must not be None.")
+        import matplotlib.lines as mlines
+        # mpl 1.3 requires us to split it like this. with recent mpl,
+        # we could use the label parameter of the Line2D
+        legend_lines, legend_labels = list(), list()
+        if cmap is None:  # ... one set of lines for the colors
+            for color in sorted(colors.keys()):
+                line = mlines.Line2D([], [], linestyle="-",
+                                     color=colors[color])
+                legend_lines.append(line)
+                legend_labels.append(color)
+        if len(list(linestyles)) > 1:  # ... one set for the linestyle
+            for style, s in linestyles.items():
+                l = mlines.Line2D([], [], color='k', linestyle=s)
+                legend_lines.append(l)
+                legend_labels.append(style)
+
+    # dealing with continuous colors
+    if cmap is not None:
+        for color_value in colors.values():
+            try:
+                float(color_value)
+            except ValueError:
+                raise ValueError("If ``cmap`` is not None, the values of "
+                                 "``colors`` need to be numeric. Got " +
+                                 str(type(color_value)))
+        cmapper = getattr(plt.cm, cmap, cmap)
+        color_conds = list(colors.keys())
+        all_colors = [colors[cond] for cond in color_conds]
+        n_colors = len(all_colors)
+        color_order = np.array(all_colors).argsort()
+        color_indices = color_order.argsort()
+
+        the_colors = cmapper(np.linspace(0, 1, n_colors))
```

Comment:
maybe it would be better to ask users to give colors as 0 - 1 floats, then non-linear mapping would be possible (for example only picking values [0.3, 0.5, 0.8])? But I am still not sure if this is better than getting the colors and passing them yourself.